### ISSUE TITLE
Installation Failed - boost.python error in job umount

### ISSUE THREAD

**Issue by arcolinuxz** [calamares/calamares#1376]
ArcoLinuxD 20.5.1
ArchLinux based
Calamares 3.2.22
virtualbox 6.1.6-1

Session log : https://pastebin.com/3QqPmbq2
End of installation - error message but after reboot installation was successful.
![error](https://user-images.githubusercontent.com/36679406/79348873-dd3d5580-7f35-11ea-99e5-86e6fc3e69c8.png)

Emergency - true on umount module
![emergency-true](https://user-images.githubusercontent.com/36679406/79349325-62286f00-7f36-11ea-8d32-01ef8681492b.png)

Can you assist?

Any config in Calamares that is not correct.

https://github.com/arcolinux/arcod-calamares-config

**Comment by abucodonosor** [comment]
Well, no idea, seems to be some ArchLinux thing, I never saw nor can I reproduce such a thing. Seem like @adriaandegroot can't either.

Basically is the same as https://github.com/calamares/calamares/issues/1333 which is closed bc no one can reproduce that.

**Comment by coritanie** [comment]
Confirm exactly the same umount error. ArcoLinuxB- i3.

**Comment by abucodonosor** [comment]
Can someone change line 101 to look like this and try again?:

```
subprocess.run(["umount", "-lv", mount_point], capture_output=True)

```

**Comment by arcolinuxz** [comment]
I have installed it now 5 times with that particular code. 
All installations were successful.
Can we add something in the debug file to get more info?
It happens on 30 installations like 2-3 times.

**Comment by abucodonosor** [comment]
@arcolinuxz 

So to reproduce one need to install in a loop?

If that is the case most likely the PIPE_BUF gets full and well hitting SIGPIPE then is
very normal.

**Comment by arcolinuxz** [comment]
the loop is not needed.
it just happens like 10% of the time.
First we thought it was a VirtualBox thing.
But it did happen on real metal too.
We keep tracking it.
The current level of logs does not provide any input for you guys.
So if we can add more logs to catch what is going on that would be great.

**Comment by adriaandegroot** [comment]
Added to the milestone, but I can't reproduce it either. I don't like this module, so for the next version expect a C++-ported version that tries to capture more output.

**Comment by adriaandegroot** [comment]
`subprocess.run()` is Python 3.5 or later, btw, so that can run into problems if we have users on the minimum-acceptable-version (Python 3.3).

I've added something equivalent just now (needs testing) which just ignores output from umount, but does consume it. That might prevent SIGPIPE.

**Comment by abucodonosor** [comment]
@adriaandegroot I know it is python >= 3.5, but since he runs 3.8 it was just to test whatever that works out that way.

**Comment by arcolinuxz** [comment]
I have used the Calamares 3.2.23 - not sure if above code is in there but this is the result on ArcoLinuxD

It is installed and will reboot without issues.

![error1](https://user-images.githubusercontent.com/36679406/80306536-36be4380-87c4-11ea-8604-6945bf179db5.jpg)
![error2](https://user-images.githubusercontent.com/36679406/80306538-39b93400-87c4-11ea-900e-d84e96bc69dd.jpg)

**Comment by arcolinuxz** [comment]
![test](https://user-images.githubusercontent.com/36679406/80864986-9ca84080-8c86-11ea-96ef-dbb974fd5f5d.png)
Can it be that the variable is in some cases just empty - I see the text "none" after inserting the code to print lst
This happens only occasionally. 4 out of 17 isos

**Comment by samurailostinjapan** [comment]
I added the following line as requested and got it to fail
subprocess.run(["umount", "-lv", mount_point], capture_output=True)
![error with capture_output=True](https://user-images.githubusercontent.com/10843832/81038691-e6d92e00-8ee1-11ea-9046-8849739f50dd.png)


Session log
https://pastebin.com/0yXAfzaJ

I also noticed during my testing that if the installation fails and before reboot - that it is still mounted  in gparted and gnome-disk
![aftererrorbeforerebootmount](https://user-images.githubusercontent.com/10843832/81038594-821dd380-8ee1-11ea-8ed4-45ee8f9c8d27.png)

**Comment by adriaandegroot** [comment]
> I added the following line as requested and got it to fail
> subprocess.run(["umount", "-lv", mount_point], capture_output=True)
> ![error with capture_output=True](https://user-images.githubusercontent.com/10843832/81038691-e6d92e00-8ee1-11ea-9046-8849739f50dd.png)

Unfortunately, that error message only tells you that the Python code is now bad. `capture_output` is a Python 3.7 addition to the API for the *Popen* object, see

https://docs.python.org/3/library/subprocess.html#subprocess.run

It looks like your system has an earlier Python version -- which is fine, because Calamares is expected to work with Python 3.3 and later. The right / better fix might be in Calamares master c59af8881cfe787a185c0b01682e7f166dd54aff , which does roughly the same, with an older API. In that change I'm assuming that the output from umount is not huge.

**Comment by adriaandegroot** [comment]
Closing this one as it's reported fixed (in master) by @arcolinuxz through other channels.
